src/main/resources/twilio-node/operation.mustache

```diff
@@ -49,23 +53,61 @@
         headers['Content-Type'] = '{{{mediaType}}}'
     {{/-first}}
     {{/consumes}}
-
     {{#headerParams}}
         if (params.{{paramName}} !== undefined) headers['{{{baseName}}}'] = params.{{paramName}};
     {{/headerParams}}
-{{/hasParams}}
+    {{/hasParams}}
 
         let operationVersion = {{#parentResourceName}}this._version{{/parentResourceName}}{{^parentResourceName}}version{{/parentResourceName}},
             operationPromise = operationVersion.{{vendorExtensions.x-name-lower}}({ uri: this._uri, method: '{{httpMethod}}'{{#hasParams}}{{#bodyParam}}, body: params{{/bodyParam}}{{^bodyParam}}, params: data{{/bodyParam}}, headers{{/hasParams}} });
-
-        {{^vendorExtensions.x-is-delete-operation}}
+        {{^vendorExtensions.x-is-list-operation}}{{^vendorExtensions.x-is-delete-operation}}
         operationPromise = operationPromise.then(payload => new {{instanceName}}(operationVersion, payload{{#resourcePathParams}}, this._solution.{{paramName}}{{/resourcePathParams}}));
-        {{/vendorExtensions.x-is-delete-operation}}
+        {{/vendorExtensions.x-is-delete-operation}}{{/vendorExtensions.x-is-list-operation}}
+        {{#vendorExtensions.x-is-list-operation}}
+        operationPromise = operationPromise.then(payload => new {{returnType}}(operationVersion, payload, this._solution));
+        {{/vendorExtensions.x-is-list-operation}}
+
+        operationPromise = operationVersion.isCallbackFunction(operationPromise,callback);
+        return operationPromise;
 
-        if (typeof callback === 'function') {
-            operationPromise = operationPromise
-                .then(value => callback(null, value))
-                .catch(error => callback(error));
+    {{#vendorExtensions.x-is-list-operation}}
+    }
+    instance.each = instance._version.each;
+    instance.list = function list(params?: any, callback?: any): Promise<{{instanceName}}[]> {
+        if (typeof params === 'function') {
+            callback = params;
+            params = {};
+        } else {
+            params = params || {};
         }
+        let allResources = [];
+        params.callback = function (resource, done) {
+            allResources.push(resource);
+            if (typeof params.limit !== 'undefined' && allResources.length === params.limit) {
+                done();
+            }
+        };
+        let operationPromise = new Promise((resolve, reject) => {
+            params.done = function (error) {
+                if (typeof error === 'undefined') {
+                    resolve(allResources);
+                } else {
+                    reject(error);
+                }
+            };
+        });
```

Comment:
Can this logic be moved to a helper like the `each` call?

Reply:
Will see if I can do it with Generics as the return type is specific unlike each